trailblazer/store/base.py

```diff
@@ -29,24 +30,54 @@ def get_job_query_with_name_and_count_labels(self) -> Query:
             func.count(Job.id).label("count"),
         )
 
-    def get_analyses_query_by_search_term_and_is_visible(
-        self,
-        search_term: str | None = None,
-        is_visible: bool = False,
-    ) -> Query:
-        """Return analyses by search term qnd is visible."""
-        if not search_term and not is_visible:
-            return
-        filter_map: dict[Callable, str | bool] | None = {
-            AnalysisFilter.FILTER_BY_IS_VISIBLE: is_visible,
-            AnalysisFilter.FILTER_BY_SEARCH_TERM: search_term,
-        }
-        filter_functions: list[Callable] = [
-            function for function, supplied_arg in filter_map.items() if supplied_arg
-        ]
-        analyses: Query = apply_analysis_filter(
-            filter_functions=filter_functions,
-            analyses=self.get_query(Analysis),
-            search_term=search_term,
-        )
-        return analyses.order_by(Analysis.started_at.desc())
+    def get_analyses_query_by_search(self, analyses: Query, search_term: str) -> Query:
+        """Return analyses by search term."""
+        if search_term:
+            analyses: Query = apply_analysis_filter(
+                filter_functions=[AnalysisFilter.FILTER_BY_SEARCH_TERM],
+                analyses=analyses,
+                search_term=search_term,
+            )
+        return analyses
+
+    def get_analyses_query_by_pipeline(self, pipeline: str) -> Query:
+        """Return analyses by pipeline."""
+        analyses: Query = self.get_query(Analysis)
+        # Group existing variants of balsamic
+        if pipeline == "balsamic":
```

Comment:
Because we have multiple variants of balsamic (balsamic, balsamic-umi etc). So retrieving analyses for balsamic requires doing a text search (expensive) against the data analysis column. We cannot do exact match

Reply:
Makes sense 👍 . Is this the desired behavior - searching for balsamic and getting all balsmaic pipelines?

Reply:
Yes, in cigrid we have a single tab for balsamic and the user expects analyses for all variants of balsamic to be returned. Maybe we should rethink this, but I'll leave it for now. I think there are some larger changes coming up for trailblazer, so we will revisit it